pyccel/codegen/printing/fcode.py

```diff
@@ -640,6 +640,12 @@ def _print_NumpyMatmul(self, expr):
         a_code = self._print(expr.a)
         b_code = self._print(expr.b)
 
+        if expr.rank == 0:
+            if isinstance(expr.a.dtype, NativeBool):
+                a_code = self._print(PythonInt(expr.a))
+            if isinstance(expr.b.dtype, NativeBool):
+                b_code = self._print(PythonInt(expr.b))
```

Comment:
I assume this is done to avoid an error about assigning a logical to an integer? But this compiler error only occurs due to issue #853 so I don't think these lines are helpful